File: solutions/3158. Find the XOR of Numbers Which Appear Twice/3158.cpp
// Time complexity: O(n)
// Space complexity: O(n)
class Solution {
 public:
  int duplicateNumbersXOR(const vector<int>& nums) {
    unordered_map<int, int> count;
    int ans = 0;

    for (const int num : nums)
      ++count[num];

    for (const auto& [num, freq] : count)
      if (freq == 2)
        ans ^= num;

    return ans;
  }
};

File: solutions/3158. Find the XOR of Numbers Which Appear Twice/3158.py
# Time complexity: O(n)
# Space complexity: O(n)
class Solution:
  def duplicateNumbersXOR(self, nums: List[int]) -> int:
    count = Counter(nums)
    ans = 0

    for num, freq in count.items():
      if freq == 2:
        ans ^= num

    return ans    